client/e2e/src/add-note.e2e-spec.ts

```diff
@@ -0,0 +1,41 @@
+import { AddNotePage, TestNote } from './add-note.po';
+import { by, element, browser, protractor } from 'protractor';
+import { E2EUtil } from './e2e.util';
+
+
+describe('Add note:', () => {
+  let page: AddNotePage;
+  const EC = protractor.ExpectedConditions;
+
+  beforeEach(() => {
+    page = new AddNotePage();
+    page.navigateTo();
+  });
+
+  it('Should have the correct title', () => {
+    expect(page.getTitle()).toEqual('New Note');
+  });
+
+  it('Should enable and disable the add note button', async () => {
+    expect(element(by.buttonText('ADD NOTE')).isEnabled()).toBe(false);
+    await page.typeInput('bodyField', 'test');
+
+    expect(element(by.buttonText('ADD NOTE')).isEnabled()).toBe(true);
+
+    await page.typeInput('bodyField', E2EUtil.randomText(250));
+    expect(element(by.buttonText('ADD NOTE')).isEnabled()).toBe(false);
+  });
+
+  it('Should add a new note and go to the right page', async () => {
+    const note: TestNote = {
+      body: E2EUtil.randomText(10)
+    };
+
+    await page.addNote(note);
```

Comment:
I'd recommend clicking the button instead of calling page.addNote directly :) That way you see that the button works!

Reply:
You even have a method to do that in add-note.po.ts already

Reply:
Oops, never mind, you're a method of `AddNotePage`—I thought you were calling a method of `AddNoteComponent` named `addNote` instead of going through the UI to do it. Pardon me! 🙇🏻‍♂️ My bad!